src/blockchain/mod.rs

```diff
@@ -500,3 +505,197 @@ impl<K: KvStore> Blockchain for KvStoreChain<K> {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{crypto, db};
+    use crate::core;
+    use crate::crypto::SignatureScheme;
+    use crate::core::{Address, TransactionData};
+
+    const DEFAULT_DIFFICULTY: u32 = 0x01ffff00;
+
+    #[test]
+    fn test_empty_chain_should_have_genesis_block() -> Result<(), BlockchainError> {
+        let chain = KvStoreChain::new(db::RamKvStore::new(), None)?;
```

Comment:
I would suggest to pass genesis block instad of "None", and then get_block(0) and then assert_eq!(first_block, genesis_block);

Reply:
No i like this test. Compare the hashes then :)